### DISCUSSION TITLE
MultivariateGrouper with FEAT_STATIC_CAT

### DISCUSSION THREAD

**Discussion by sege01** [awslabs/gluonts#1692] [Q&A] [answered]
Hello,

I had a question regarding MultivariateGrouper.
Is it possible to use feat_static_cat features with multivariate grouper?
As in the method `prepare_train_data(self, dataset: Dataset)` of class `MultivariateGrouper`
it is set as below to [0]:
` grouped_data[FieldName.FEAT_STATIC_CAT] = [0]`

I have a dataset that looks like below:
![image](https://user-images.githubusercontent.com/85530006/133657957-f7bd4b86-6a30-4138-8fc2-2865bcd4f41f.png)
where each row represents a time series associated with static categorical features.
How should be the data formatted using multivariate grouper if I need a multivariate dataset which contains static categorical features as well? 
@mbohlkeschneider @lostella Could you please help with this?

Thank you!

**Comment by mbohlkeschneider** [comment, marked as answer]
Hi @sege01, 

The multivariate grouper will turn all your marginal timeseries in a multivariate time series. The static cat feature has no meaning in this case (because it is always the same).

Hope that helps with the understanding.

**Comment by sege01** [comment]
Thank you @mbohlkeschneider .

Also, how can be the dataset like m5 forecasting
https://github.com/awslabs/gluon-ts/blob/master/examples/m5_gluonts_template.ipynb
transformed in a gluonts format that can produce forecasts with multivariate models? For example  DeepVar i.e multivariate variant of DeepAR.
Like the dataset contains different stat cat features like item_ids, dept_ids, cat_ids, store_ids, state_ids which are not the same for all-time series.

Thanks.

**Reply by mbohlkeschneider** [reply]
Hi @sege01,

In case of DeepAR, the static cat feature signals the model which time series is drawn in training. In the case of DeepVAR, this actually does not matter if all time series are grouped into one multivariate time series (which the `MultivariateGrouper` is doing). For example consider two time series: store 1, `[0,0,0,0,0]` and store 2 `[1,1,1,1,1]`. If I put them into a multivariate time series it becomes `[[0,0,0,0,0], [1,1,1,1,1]]` which represents store 1 AND store 2. At this point, it the static cat features do not matter anymore and are implicitly encoded in the dimension of the individual time series. 

That being said: You *could* split the dataset into different multivariate groups and then those features might be relevant again, but there is no tooling in GluonTS to do that. 

Hope that clarified it.

**Comment by sege01** [comment]
Thank you so much @mbohlkeschneider for the explanation.
